subprojects/common/elastic/src/main/kotlin/com/avito/android/elastic/RoundRobinService.kt

```diff
@@ -0,0 +1,22 @@
+package com.avito.android.elastic
+
+import okhttp3.ResponseBody
+import retrofit2.Call
+import java.net.URL
+
+internal class RoundRobinService(
+    factory: (endpoint: URL) -> ElasticService,
+    endpoints: List<URL>
+) : ElasticService {
+
+    private val roundRobin =
+        RoundRobinIterable(endpoints.map { endpoint -> lazy { factory(endpoint) } })
+
+    override fun log(indexPattern: String, date: String, params: Map<String, String>): Call<ResponseBody> {
+        return roundRobin.next().value.log(
```

Comment:
As before, I don't understand why we switch to the next destination on each invocation but not after error.

Reply:
i will write a retry interceptor for that